dropshot/src/server.rs

```diff
@@ -281,9 +304,11 @@ impl<C: ServerContext> InnerHttpServerStarter<C> {
         private: C,
         log: &Logger,
         handler_waitgroup_worker: waitgroup::Worker,
-    ) -> Result<InnerHttpServerStarterNewReturn<C>, hyper::Error> {
-        let incoming = AddrIncoming::bind(&config.bind_address)?;
-        let local_addr = incoming.local_addr();
+    ) -> Result<InnerHttpServerStarterNewReturn<C>, std::io::Error> {
+        let std_listener = std::net::TcpListener::bind(&config.bind_address)?;
+        std_listener.set_nonblocking(true)?;
+        let incoming = TcpListener::from_std(std_listener)?;
+        let local_addr = incoming.local_addr()?;
 
```

Comment:
can you talk about these changes or explain in a comment?

Reply:
This is because the whole `AddrStream` thing went away, since it offered basically nothing over a `TcpStream`. This conversion dance allows binding to happen synchronously, since `tokio::net::TcpListener::bind` is an `async fn` since it allows a string argument that will trigger an async DNS lookup.

Reply:
Ah, it's also commented about originally in L625 of this file, where the same thing is done to build a TLS acceptor.

Reply:
Interesting about Tokio's `TcpListener::bind` being an async fn. Is making a std `TcpListener` and then converting it to Tokio the general recommendation? (I've used a similar pattern for Tokio's processes so it's not a huge surprise to me if it is.)

Could you add a comment here explaining this?

Reply:
Yea, I can copy over the comment from the other line this pattern is used.